packages/block-editor/src/store/private-selectors.js

```diff
@@ -81,6 +84,29 @@ export const isBlockSubtreeDisabled = ( state, clientId ) => {
 	return getBlockOrder( state, clientId ).every( isChildSubtreeDisabled );
 };
 
+/**
+ * Determines if a container (clientId) allows insertion of blocks, considering contentOnly mode restrictions.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId The client ID of the container block.
+ *
+ * @return {boolean} Whether the container allows insertion.
+ */
+export function isContainerInsertableToInWriteMode( state, clientId ) {
```

Comment:
This function deliberately doesn't check whether we _are_ in write mode or not, it just checks the conditions for a container to be insertable to if it's in write mode. It only exists so we can use it in both `canInsertBlockTypeUnmemoized` and `canRemoveBlock` so that they don't fall out of sync. If it's possible to add a block, it should always be possible to remove it.